ROPC remediation - configuration/index.md, use-http-context.md, new include file (#33858)
dotnet/AspNetCore.Docs: aspnetcore/fundamentals/configuration/index.md

* Never store passwords or other sensitive data in configuration provider code or in plain text configuration files. The [Secret Manager](xref:security/app-secrets) tool can be used to store secrets in development.
* Don't use production secrets in development or test environments.
* Specify secrets outside of the project so that they can't be accidentally committed to a source code repository.

> * Avoid the use of passwords in production apps; for more information, see [Secure authentication flows](xref:security/index#secure-authentication-flows).

Reviewer: I don't understand this. Production apps need passwords. I think this is saying:
Never store passwords or other sensitive data in configuration provider code or in plain text configuration files. 

But we've already said that. But the link to secure auth flow is good.

Author: > I don't understand this. Production apps need passwords. I think this is saying: Never store passwords or other sensitive data in configuration provider code or in plain text configuration files.
> 
> But we've already said that. But the link to secure auth flow is good.

What then would you propose for brief text that says avoid username/password when you can, use managed identities when you can? Like this section of https://learn.microsoft.com/en-us/entra/identity-platform/authentication-flows-app-scenarios#web-app-that-signs-in-a-user:

>Though we don't recommend that you use it, the [username/password flow](https://learn.microsoft.com/en-us/entra/identity-platform/scenario-desktop-acquire-token-username-password) is available in public client applications. This flow is still needed in some scenarios like DevOps.

> Using the username/password flow constrains your applications. For instance, applications can't sign in a user who needs to use multifactor authentication or the Conditional Access tool in Microsoft Entra ID. Your applications also don't benefit from single sign-on. Authentication with the username/password flow goes against the principles of modern authentication and is provided only for legacy reasons.

> * Avoid the use of passwords in production apps; for more information, see [Secure authentication flows](xref:security/index#secure-authentication-flows).

Author: ```suggestion
* Production apps should use the most secure authentication flow available. For more information, see [Secure authentication flows](xref:security/index#secure-authentication-flows).
```
